### ISSUE TITLE
New version not working with OpenWeatherMap

### ISSUE THREAD

**Issue by zendiik** [pkissling/clock-weather-card#288]
**Checklist:**

- [x] I updated to the latest version available
- [x] I cleared the cache of my browser
- [x] I read the [FAQ](https://github.com/pkissling/clock-weather-card#faq) and the issue I'm having was not mentioned there.

**Release with the issue:**

**Last working release (if known):**
2.1.5

**Browser and Operating System:**
Firefox Developer 120, Arch Linux

**Description of problem:**
The card doesn't show up on the dashboard. 

**Javascript errors shown in the web inspector (if applicable):**

```
Uncaught (in promise) Error: Weather entity "weather.openweathermap" does not support daily forecasts.
```
![image](https://github.com/pkissling/clock-weather-card/assets/10212768/ccfacf6b-6055-46be-b45a-94dcb9fc5a07)

**Comment by swampylee** [comment]
Yep - even trying to downgrade back to previous version (and clearing cache) doesn't seem to fix it.

**Comment by pkissling** [comment]
hi there, what is your weather entity set to? daily or hourly? 
<img width="452" alt="image" src="https://github.com/pkissling/clock-weather-card/assets/33731393/d2ae3263-c54f-4210-adb0-eca1f119e059">

**Comment by zendiik** [comment]
I have hourly
![image](https://github.com/pkissling/clock-weather-card/assets/10212768/d9ba377d-6a14-4ddf-9c3d-8ca5cf22769e)

**Comment by pkissling** [comment]
the error message is "weather entity does not support daily forecasts". can you set the entity to to daily then?

**Comment by swampylee** [comment]
If I change to daily in OpenWeatherMap that integration fails. Setting it back to hourly and it's fine (card still doesn't load though)

**Comment by pkissling** [comment]
i did some analysis to understand why this issue occurs:

until 2.1.6 clock-weather-card would always try to read first the `forecast` attribute available in the weather entity (despite the weather provider supporting the new `weather.get_forecast` and only if `forecast` was not available, it would use `weather.get_forecast` to obtain `forecast`s
<img width="1028" alt="image" src="https://github.com/pkissling/clock-weather-card/assets/33731393/33d63784-a456-4623-a28e-338447d5185c">
in order to fix #218 i implemented a change (released with 2.1.6) so that clock-weather-card would rely on `supported_features` in the weather entity either subscribe to forecast updates via `weather.get_forecast` or read it from `forecast` attribute in the weather entity.

now there is a catch with openweathermap:
in the free tier you are only able to subscribe to hourly forecasts. but with the default setup of `clock-weather-card`, you want to display the weather daily. so clock-weather-card will try to subscribe to daily updates, which is not supported by openweathermap. this is why you see the error

to fix the issue i added some logik to make clock-weather-card fall back to hourly forecast subscription in case daily is not supported. i think this should fix your issue.
